test/Microsoft.AspNet.Mvc.Core.Test/ActionResults/CreatedAtActionResultTests.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.PipelineCore.Collections;
+using Microsoft.AspNet.Routing;
+using Microsoft.AspNet.Testing;
+using Microsoft.Framework.DependencyInjection;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class CreatedAtActionResultTests
+    {
+        [Fact]
+        public async Task CreatedAtActionResult_ReturnsStatusCode_SetsLocationHeader()
+        {
+            // Arrange
+            var appRoot = "";
+            var expectedUrl = "testAction";
+            var response = GetMockedHttpResponseObject();
+            var httpContext = GetHttpContext(appRoot, response);
+            var actionContext = GetActionContext(httpContext);
+            var urlHelper = GetMockUrlHelper(expectedUrl);
+
+            // Act
+            var result = new CreatedAtActionResult(expectedUrl, null, null, null);
+            result.UrlHelper = urlHelper;
+            await result.ExecuteResultAsync(actionContext);
+
+            // Assert
+            Assert.Equal(201, response.StatusCode);
+            Assert.Equal(expectedUrl, response.Headers["Location"]);
+        }
+
+        [Fact]
+        public async Task CreatedAtActionResult_ThrowsOnNullUrl()
+        {
+            // Arrange
+            var appRoot = "";
+            var response = GetMockedHttpResponseObject();
+            var httpContext = GetHttpContext(appRoot, response);
+            var actionContext = GetActionContext(httpContext);
+            var urlHelper = GetMockUrlHelper(returnValue: null);
+
+            var result = new CreatedAtActionResult(null, null, null, null);
+            result.UrlHelper = urlHelper;
+
+            // Act & Assert
+            await ExceptionAssert.ThrowsAsync<InvalidOperationException>(
+                async () => await result.ExecuteResultAsync(actionContext), 
+            "No route matches the supplied values.");
+        }
+
+        private static HttpResponse GetMockedHttpResponseObject()
+        {
+            var stream = new MemoryStream();
+            var httpResponse = new Mock<HttpResponse>();
+            httpResponse.SetupProperty(o => o.StatusCode);
+            httpResponse.Setup(o => o.Headers).Returns(
+                new HeaderDictionary(new Dictionary<string, string[]>()));
+            httpResponse.SetupGet(o => o.Body).Returns(stream);
+            return httpResponse.Object;
+        }
+
+        private static ActionContext GetActionContext(HttpContext httpContext)
+        {
+            var routeData = new RouteData();
+            routeData.Routers.Add(new Mock<IRouter>().Object);
```

Comment:
This is fine as is, but you could do `Mock.Of<IRouter>()` if you don't care about setting up the Mocked type.